### DISCUSSION TITLE
Номер телефона как ссылка

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#237] [Как это сделать?] [answered]
Подскажите пожалуйста, как сделать значение с номером телефона - кликабельным, чтобы на телефоне открывалась звонилка по клику на номер?
Преобразование в URL не знаю как применить.

**Comment by erpic-ru** [comment, marked as answer]
Поле с телефоном должно быть нередактируемое или редактируемым с указанием, что клик как URL. Код действия по клику:

<img width="1152" alt="Снимок экрана 2022-10-21 в 14 28 45" src="https://user-images.githubusercontent.com/55755565/197185886-e6dcd3c3-3fcf-479e-b9ee-0eafc69e8b5d.png">

**UPD  [Полунин]:** Клик как URL не читается на мобиле (посмотрим на сл неделе).

**Reply by ttm-alexeypolunin** [reply]
Можно еще сделать рядом поле с аналогичным образом сгенерированным значением вида `tel:79995554389`, через setFormat text показывать нормальный телефон, поставить оформление как ссылки и через тот-же setFormat параметр hide скрывать на мобиле основное поле, а на десктопе мобильное.

**Reply by mihail-cornev** [reply]
> Можно еще сделать рядом поле с аналогичным образом сгенерированной ссылкой, через setFormat text показывать нормальный телефон, поставить оформление как ссылки и через тот-же setFormat параметр hide скрывать на мобиле основное поле, а на десктопе мобильное.

Я пробовал через setformat но оно делает ссылку вида "../Table/20/46/37369857197"
Я только неделю как изучаю учебный курс, поэтому простите, если какие-то вещи не догоняю сразу.

**Reply by mihail-cornev** [reply]
> Поле с телефоном должно быть нередактируемое. Код действия по клику:
> 
> ```
> = :str`"<a href=" + "tel:" + $#nfv + ">"`
> ```

никак не выходит повторить то, что вы предложили ( не работает у меня.

**Reply by ttm-alexeypolunin** [reply]
Протестил и откорректировал сообщения выше.

**Reply by mihail-cornev** [reply]
> Протестил и откорректировал сообщения выше.

Спасибо, великолепно!

**Reply by mihail-cornev** [reply]
Перестала работать данная штука.

**Comment by mihail-cornev** [comment]
ничего не менял с того раза как всё работало. просто перестало работать.
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/113771444/7dc2b107-5b27-42c2-80ec-6c23c5000d1b)

**Reply by ttm-alexeypolunin** [reply]
Посмотрим

**Reply by ttm-alexeypolunin** [reply]
у меня на мак и хром и фаерфокс честно пытаются отработать tel: и открывают facetime

**Reply by mihail-cornev** [reply]
снова заработало, но я так и не понял с чем было связано. возможно на моём ноуте что-то, хотя на других сайтах эта штука работала.

**Reply by mihail-cornev** [reply]
снова перестало работать. нашёл вот такую инфу.
https://9to5google.com/2023/08/30/google-chrome-click-to-call-feature-removed/
